### ISSUE TITLE
Access: walletconnect: android com: 1.31.0 Compile duplicate class error May I ask how to solve it

### ISSUE THREAD

**Issue by vsmzz** [WalletConnect/WalletConnectKotlinV2#1355]
implementation(platform("com.walletconnect:android-bom:1.31.0"))
    implementation("com.walletconnect:android-core")
    implementation("com.walletconnect:web3modal")
    gradle：
     id("com.android.application") version "7.4.2" apply false
    id("org.jetbrains.kotlin.android") version "1.8.21" apply false
    
    
    
Compilation error：
Duplicate class org.bouncycastle.LICENSE found in modules bcprov-jdk15on-1.68 (org.bouncycastle:bcprov-jdk15on:1.68) and bcprov-jdk18on-1.77 (org.bouncycastle:bcprov-jdk18on:1.77)

2 files found with path 'org/bouncycastle/x509/CertPathReviewerMessages_de.properties' from inputs:

**Comment by TalhaAli00** [comment]
@vsmzz 

can you try adding this to your root build.gradle file and let us know if it resolves the issue for you?

```kotlin
configurations.configureEach { 
    resolutionStrategy.eachDependency {
        if (requested.group == "org.bouncycastle" && requested.name == "bcprov-jdk15on") {
            useTarget("org.bouncycastle:bcprov-jdk18on:1.77")
        }
    }
}
```

**Comment by vsmzz** [comment]
> @vsmzz
> 
> 您可以尝试将其添加到您的根 build.gradle 文件中，并让我们知道它是否可以解决您的问题吗？
> 
> ```kotlin
> configurations.configureEach { 
>     resolutionStrategy.eachDependency {
>         if (requested.group == "org.bouncycastle" && requested.name == "bcprov-jdk15on") {
>             useTarget("org.bouncycastle:bcprov-jdk18on:1.77")
>         }
>     }
> }
> ```
Thank you very much, this is very useful

**Comment by TalhaAli00** [comment]
@vsmzz We've just released BOM 1.31.1 which includes a fix for the dependency issue
